packages/beacon-state-transition/src/cache/epochContext.ts

```diff
@@ -469,7 +469,8 @@ export class EpochContext {
 
   getBeaconProposer(slot: Slot): ValidatorIndex {
     const epoch = computeEpochAtSlot(slot);
-    if (epoch !== this.currentShuffling.epoch) {
+    // disallow retrieving proposer for past epochs and epochs more than 1 epoch in the future
+    if (epoch < this.currentShuffling.epoch || epoch > this.currentShuffling.epoch + 1) {
```

Comment:
why do we need this change? state-transition only caches proposers for current epoch, not next epoch

Reply:
Without this extra check, if proposers for more than 1 epoch is requested, the proposer for current epoch is returned. That is because of `return this.proposers[slot % SLOTS_PER_EPOCH]` on line 478. 

This checks ensures an error message is returned instead when proposers for more than 1 epoch ahead is requested.

Reply:
I mean the old check is more strict and we should keep it. Do we have any consumers calling this function with `epoch !== this.currentShuffling.epoch`?

Reply:
You correct. The change was not needed. Reverted.